Modules/Devices/test/Devices.Domain.Tests/Identities/CancelDeletionProcessTests.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Backbone.BuildingBlocks.Domain;
+using Backbone.Modules.Devices.Domain.Entities.Identities;
+using Backbone.Tooling;
+using FluentAssertions;
+using Xunit;
+
+namespace Backbone.Modules.Devices.Domain.Tests.Identities;
+public class CancelDeletionProcessTests
+{
+    [Fact]
+    public void Should_cancel_deletion_process_during_grace_period()
+    {
+        // Arrange
+        var currentDate = SystemTime.UtcNow.Date;
+        SystemTime.Set(currentDate);
+
+        var activeIdentity = TestDataGenerator.CreateIdentity();
+        var activeDevice = new Device(activeIdentity);
+        activeIdentity.Devices.Add(activeDevice);
+
+        activeIdentity.StartDeletionProcessAsOwner(activeDevice.Id);
+        SystemTime.Set(currentDate.AddDays(5));
+
+        // Act
+        activeIdentity.CancelDeletionProcess(activeDevice.Id);
+
+        // Assert
+        var result = activeIdentity.DeletionProcesses.Where(dp => dp.Status == DeletionProcessStatus.Cancelled).ToList();
```

Comment:
Here you could use the deletion process you saved as described in my previous comment